treecorr/catalog_formats.py

```diff
@@ -0,0 +1,368 @@
+        index in the file.
+
+        Parameters
+        ----------
+        ext: int/str
+            name or index to search for
+
+        Returns
+        -------
+        bool
+            Whether the extension exists
+        """
+        return ext in self.file
+
+    def __enter__(self):
+        # Context manager, enables "with FitsReader(filename) as f:"
+        return self
+
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        # Context manager closer - we just close the file at the end,
+        # regardless of the error
+        self.file.close()
+
+    def check_valid_ext(self, ext):
+        """Check if an extension is valid for reading, and raise ValueError if not.
+
+        The ext must both exist and be a table (not an image)
+
+        Parameters
+        ----------
+        ext: int/str
+            The extension to check
+        """
+        import fitsio
+
+        if ext not in self:
+            raise ValueError("Invalid ext={} for file {} (does not exist)".format(
+                             ext, self.file_name))
+
+        if not isinstance(self.file[ext], fitsio.hdu.TableHDU):
+            raise ValueError("Invalid ext={} for file {} (Not a TableHDU)".format(
+                             ext, self.file_name))
+
+    @staticmethod
+    def choose_extension(config, name, num, default=None):
+        """Select an extension name or index from a configuration.
+
+        If no key is found or default supplied, fall back to the first FITS extension.
+
+        Parameters
+        ----------
+        config: dict
+            config to choose from
+        name: str
+            parameter name to get
+        num: int
+            if the value is a list, which item to get
+        default: int/str
+            optional, the fall-back if not found
+
+        Returns
+        -------
+        ext: int/str
+            The value that can be used to look up the extension
+        """
+
+        # get the value as a string - if it's actually an int
+        # we will convert below
+        ext = get_from_list(config, name, num, str)
+
+        # allow the user to use the old _hdu format names, but warn them
+        # that these are deprecated.
+        if ext is None:
+            ext = get_from_list(config, name.replace('ext', 'hdu'), num, str)
+
+            if ext is not None:
+                warnings.warn("Parameters ending in 'hdu' are deprecated and "
+                              "should now be renamed to end in 'ext' instead",
+                              FutureWarning)
+
+        # If not found, use the default if present, otherwise the global
+        # default of 1
+        if ext is None:
+            if default is None:
+                ext = 1
+            else:
+                ext = default
+
+        # FITS extensions can be indexed by number or
+        # string.  Try converting to an integer first
+        try:
+            ext = int(ext)
```

Comment:
Seems unlikely to be an issue.  And elsewhere I think you said if people use string numbers for extension names, they deserve what they get.  But I think we should only bother with this -> int try `if ext not in self`.